internal/plan/plan.go

```diff
@@ -184,6 +184,61 @@ func (s *Service) Equal(other *Service) bool {
 	return reflect.DeepEqual(s, other)
 }
 
+// GetCommand returns the service command as a stream of strings.
+// It adds the arguments in cmdArgs to the command if no default
+// argument is specified in [ ... ] group in the plan.
+// If default arguments are specified in the plan, the default
+// arguments are added to the command if cmdArgs is empty.
+// If not empty, the default arguments are overrided by arguments in cmdArgs.
+func (s *Service) GetCommand(cmdArgs []string) ([]string, error) {
```

Comment:
Nit: I'd probably prefer this as a stand-alone helper function (not a method), so it's more explicit about the dependencies, and so callers don't necessarily need a `Service` instance lying around. What about this:

```go
func CommandArgs(command string, additionalArgs []string) ([]string, error) { ... }
```